src/snowflake/snowpark/async_job.py

```diff
@@ -206,45 +278,84 @@ def _table_result(
             )
 
     @experimental(version="0.10.0")
-    def result(self) -> Union[List[Row], "pandas.DataFrame", Iterator[Row], int, None]:
+    def result(
+        self,
+        result_type: Optional[
```

Comment:
What if ``self.result_type`` and this ``result_type`` are different?

Reply:
What `rows` would be if I do
```python
aj = df.to_pandas(block=False)
rows = aj.result("rows")
```

Reply:
It will return a list of rows. This argument provides users with overwriting the original result type. It's documented in the docstring and you can find something similar in the test.